oak_restricted_kernel_sdk/src/dice.rs

```diff
@@ -22,27 +22,29 @@ use oak_restricted_kernel_interface::DICE_DATA_FD;
 use p256::ecdsa::SigningKey;
 use zerocopy::{AsBytes, FromZeroes};
 
+lazy_static::lazy_static! {
+    static ref DICE_WRAPPER: anyhow::Result<DiceWrapper> = {
+        let dice_data = get_restricted_kernel_dice_data()?;
+        let encryption_key = EncryptionKeyProvider::try_from(&dice_data)?;
+        let signing_key = SigningKey::from_slice(
+            &dice_data.application_private_keys.signing_private_key[..P256_PRIVATE_KEY_SIZE],
+        )
+        .map_err(|error| anyhow::anyhow!("couldn't deserialize signing key: {}", error))?;
+        let evidence = dice_data.evidence;
+        Ok(DiceWrapper {
+            evidence,
+            encryption_key,
+            signing_key,
+        })
+    };
+}
+
```

Comment:
We removed existing uses of lazy_static previously, so I would prefer not to add it back in. Lazy loading makes reasoning about state and troubleshooting launch issues a bit more difficult. Also, since the lazy loading can fail, it means that we always have to deal with the result rather than handling the result once and then just dealing with the actual value the rest of the fime.

Which OnceCell did you use? We have implemented https://github.com/project-oak/oak/blob/main/oak_core/src/sync.rs#L28 which should be `Sync`.

Reply:
> We removed existing uses of lazy_static previously, so I would prefer not to add it back in. Lazy loading makes reasoning about state and troubleshooting launch issues a bit more difficult. Also, since the lazy loading can fail, it means that we always have to deal with the result rather than handling the result once and then just dealing with the actual value the rest of the fime.

I think we still use it one of the enclave apps, but good reasons not to use it imo.

> Which OnceCell did you use? We have implemented https://github.com/project-oak/oak/blob/main/oak_core/src/sync.rs#L28 which should be `Sync`.

Oh that's a good one, was unaware of this. Previously I just attempted to use the core variant of `OnceCell`. I updated this code to use our `OnceCell`. In this case behavior stays the same, as I actually think that the behavior of `lazy_static` is precisely what we want here. — But this way the intention is more clearly reflected in the code.